### PR TITLE
Add column families support

### PR DESCRIPTION
**Pull request by de-luca**
Hi,

This PR adds support for [column families](https://www.cockroachlabs.com/docs/stable/column-families).

### CODE REVIEW

File: cockroachdb-dialect/src/main/kotlin/com/faire/sqldelight/dialects/cockroachdb/grammar/CockroachDB.bnf [hunk 1]
```diff
@@ -112,7 +112,9 @@ table_constraint ::= [ CONSTRAINT ansi_identifier ] (
   override = true
 }
 
-column_constraint ::= [ CONSTRAINT ansi_identifier ] (
+table_family_constraint ::= 'FAMILY' ansi_identifier LP ansi_column_name ( COMMA ansi_column_name ) * RP
```
Comment (Adriel-M) [inline cockroachdb-dialect/src/main/kotlin/com/faire/sqldelight/dialects/cockroachdb/grammar/CockroachDB.bnf:115]: It looks like `ansi_identifier` is optional

https://www.cockroachlabs.com/docs/v23.2/sql-grammar#family_def

To verify, can you add an integration test for this?

File: cockroachdb-dialect/src/main/kotlin/com/faire/sqldelight/dialects/cockroachdb/grammar/CockroachDB.bnf [hunk 2]
```diff
@@ -127,6 +129,8 @@ column_constraint ::= [ CONSTRAINT ansi_identifier ] (
   override = true
 }
 
+alter_table_add_column_family_constraint ::= [ CREATE [ IF NOT EXISTS ] ] 'FAMILY' ansi_identifier
```
Comment (Adriel-M) [inline cockroachdb-dialect/src/main/kotlin/com/faire/sqldelight/dialects/cockroachdb/grammar/CockroachDB.bnf:132]: Some are optional here though: https://www.cockroachlabs.com/docs/v23.2/sql-grammar#col_qualification

**Review by Adriel-M — COMMENTED** [review]
Thanks for another contribution 🤩 

Just a bunch of extra stuff since the family name can be optional

File: cockroachdb-dialect/src/main/kotlin/com/faire/sqldelight/dialects/cockroachdb/grammar/CockroachDB.bnf (hunk 2 above)
Comment (de-luca) [reply cockroachdb-dialect/src/main/kotlin/com/faire/sqldelight/dialects/cockroachdb/grammar/CockroachDB.bnf]: I split the branches, it should match now

**Review by Adriel-M — APPROVED** [review]

**Comment by MarvatheLarva** [comment]
Hello & Thanks for the great work !

Is there any release plan soon ?
We would love to have this `FAMILY ` keyword :)

**Comment by Adriel-M** [comment]
@MarvatheLarva I'll look into doing that sometime this week. Current state of things is that the postgres dialect is broken from https://github.com/cashapp/sqldelight/releases/tag/2.0.1 but I'll most likely just start using a pinned SNAPSHOT version of sqldelight.

**Pull request merged**